cmd/rancher-desktop-guestagent/main.go

```diff
@@ -75,29 +80,60 @@ func main() {
 
 	group, ctx := errgroup.WithContext(context.Background())
 
-	if *enableDocker {
+	if *enablePriviledService && *enableIptables {
+		log.Fatal("error enabling Privileged Service mode and iptables, you can only enable one.")
+	}
+
+	if *enablePriviledService {
+		if !*enableContainerd && !*enableDocker {
+			log.Fatal("enabling Privileged Service mode requires either Docker or Containerd enabled.")
+		}
+
+		if *enableContainerd && *enableDocker {
+			log.Fatal("enabling Privileged Service mode requires enabling either Docker or Containerd not both.")
+		}
+
 		if *vtunnelAddr == "" {
-			log.Fatal("vtunnel address must be provided when docker is enable.")
+			log.Fatal("vtunnel address must be provided when docker is enabled.")
```

Comment:
What about only `-containerd` enabled? You haven't checked `*enableDocker` by itself in this code yet.

Reply:
I think at some point in the future we should move to cobra because there is a lot of checking scenarios that are not covered.